packages/algob/src/internal/deployer.ts

```diff
@@ -104,22 +106,55 @@ export class AlgobDeployerImpl implements AlgobDeployer {
 
   async deployASA (name: string, flags: ASADeploymentFlags): Promise<ASAInfo> {
     if (this.loadedAsaDefs[name] === undefined) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
       throw new BuilderError(
         ERRORS.BUILTIN_TASKS.DEPLOYER_ASA_DEF_NOT_FOUND, {
           asaName: name
         });
     }
     this.assertNoAsset(name);
-    const asaInfo = await this.algoOp.deployASA(
-      name, this.loadedAsaDefs[name], flags, this.accountsByName, this.txWriter);
+    let asaInfo = {} as any;
+    try {
+      asaInfo = await this.algoOp.deployASA(
+        name, this.loadedAsaDefs[name], flags, this.accountsByName, this.txWriter);
+    } catch (error) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+
+      console.log(error);
+      throw new Error();
```

Comment:
don't throw empty error. You can re-throw existing:

```
throw error
```

Reply:
ok